addon/lib/shooter.js

```diff
@@ -296,14 +296,33 @@ const ShotContext = Class({
       }
       this.updateShot();
     },
-    copyLink: function () {
-      clipboard.set(this.shot.viewUrl, "text");
+    copyLink: function (url) {
+      clipboard.set(url, "text");
       notifications.notify({
         title: "Link Copied",
         text: "The link to your shot has been copied to the clipboard.",
         iconURL: self.data.url("../data/copy.png")
       });
     },
+    copyImage: function (img) {
+      clipboard.set(img, "image");
+      notifications.notify({
+        title: "Image Copied",
+        text: "Your shot has been copied to the clipboard.",
+        iconURL: self.data.url("../data/copy.png")
+      });
+    },
+    copyRich: function (url, img, title) {
+      // Use "text" instead of "html" so that pasting into a text area or text editor
+      // pastes the html instead of the plain text stripped out of the html.
+      clipboard.set(
+        `<a href="${url}"><img src="${img}" /><div>${title}</div></a>`, "text");
```

Comment:
Ok, I added code that sets the clipboard html type, too. It doesn't seem to have any effect.

I can't seem to get facebook, twitter, or gmail to do anything other than insert the raw html into the textbox. If I set the clipboard html type but not the text type, then the only thing that is pasted is whatever visible text there would have been in that html -- which is not the right thing to paste at all.